HealthyLLM/HealthyLLM/Fetcher/HealthDataFetcher.swift

```diff
@@ -8,99 +8,191 @@
             }
-            
-            return HealthData(
-                name: identifier.rawValue,
-                unit: unit.unitString,
-                values: results
-            )
         }
+
+        return HealthData(
+            name: sampleType.displayTitle,
+            unit: unit.unitString,
+            values: result
+        )
     }
     
     func fetchSleep() async -> String { "" }
     
-    func fetchWorkout(type: String) async -> [WorkoutData]? {
-        guard let identifier = workoutStringToHKWorkoutActivityType(type),
-              let result = await fetchWorkoutData(activity: identifier) else {
+    func fetchWorkout(_ healthKit: HealthKit, type: String) async -> [WorkoutData]? {
+        guard let activity = workoutStringToHKWorkoutActivityType(type),
+              let workouts = try? await healthKit.query(.workout, timeRange: .ever, limit: 10, sortedBy: [SortDescriptor(\.startDate, order: .reverse)]) else {
             return nil
         }
-        
+
+        let filtered = workouts.filter { $0.workoutActivityType == activity }
+        var result: [WorkoutData] = []
+
+        for workout in filtered {
+            var stats: [String: String] = [:]
+
+            for quantityType in workout.allStatistics.keys {
+                guard let statistics = workout.allStatistics[quantityType],
+                      let (identifier, _) = hkStringToHKQuantityTypeIdentifier(quantityType.identifier),
+                      let unit = identifier.siUnit else { continue }
+
+                let shortIdentifier = quantityType.identifier.replacingOccurrences(of: "HKQuantityTypeIdentifier", with: "")
+
+                switch quantityType.aggregationStyle {
+                case .cumulative:
+                    if let sum = statistics.sumQuantity() {
+                        let value = sum.doubleValue(for: unit)
+                        stats[shortIdentifier] = "\(value.rounded()) \(unit)"
+                    }
+                case .discreteArithmetic, .discreteTemporallyWeighted:
+                    if let average = statistics.averageQuantity() {
+                        let value = average.doubleValue(for: unit)
+                        stats[shortIdentifier] = "\(value.rounded()) \(unit)"
+                    }
+                default:
+                    continue
+                }
+            }
+
+            result.append(
+                .init(
+                    name: String(describing: workout.workoutActivityType),
+                    date: workout.startDate.formatted(date: .abbreviated, time: .shortened),
+                    duration: Duration.seconds(workout.duration).formatted(),
+                    statistics: stats
+                )
+            )
+        }
+
         return result
     }
 }
+
+// One-off query for aggregating health data
+// TODO: Create Pull Request for SpeziHealthKit
+extension HealthKit {
+    public func statisticsQuery<Sample>(
```

Comment:
@lukaskollmer do you know if `SpeziHealthKit` already provides the functionality for one-off statistics queries in some way? If not, and you think it would be useful to others as well, I'm happy to create a PR!

Reply:
the `@HealthKitStatisticsQuery` API already contains an initial attempt at making this type safe; we have different init overloads to prevent the caller from passing in invalid aggregation option combinations

Reply:
@lukaskollmer and I had a small discussion on Monday about a very simple approach to make the SwiftUI property wrappers also available outside of the SwiftUI hierarchy and usable inside modules, we might want to tackle this in the future, IMO that would solve the issue here as well. But we might want merge this first and then iterate on it later.

Reply:
Sounds good, @lukaskollmer let me know if you need some support, but for now I agree with merging first and iterating later 👍